aclk/aclk_common.c

```diff
@@ -144,14 +148,31 @@ const char *aclk_lws_wss_get_proxy_setting(ACLK_PROXY_TYPE *type)
         return proxy;
 
     if (strcmp(proxy, ACLK_PROXY_ENV) == 0) {
-        if (check_socks_enviroment(&proxy) == 0)
+        if (check_socks_enviroment(&proxy) == 0) {
+#ifdef LWS_WITH_SOCKS5
             *type = PROXY_TYPE_SOCKS5;
-        else if (check_http_enviroment(&proxy) == 0)
+            return proxy;
+#else
+            safe_log_proxy_error("socks_proxy environment variable set to use SOCKS5 proxy"
+                " but Libwebsockets used doesn't have SOCKS5 support built in. Ignoring this.",
+                proxy);
+#endif
+        }
+        if (check_http_enviroment(&proxy) == 0)
             *type = PROXY_TYPE_HTTP;
         return proxy;
     }
 
     *type = aclk_verify_proxy(proxy);
+#ifndef LWS_WITH_SOCKS5
+    if (*type == PROXY_TYPE_SOCKS5) {
+        char *log = strdupz(proxy);
+        safe_log_proxy_censor(log);
+        fatal("Config var \"" ACLK_PROXY_CONFIG_VAR "\" set to use SOCKS5 proxy but this agent "
```

Comment:
We shouldn't kill the agent, but we should drop ACLK instead.

Reply:
I changed it to behavior discussed on Slack.
Now proxy setting is checked at the beginning of the ACLK thread and will be disabled if sepcifically SOCKS5 is requested by config.

Reply:
that means we will not `fatal` after last changes same as you suggested